wallet/src/account/operations/transaction/prepare_output.rs

```diff
@@ -167,9 +160,19 @@ impl AccountHandle {
         // We might have added more unlock conditions, so we check the minimum storage deposit again and update the
         // amounts if needed
         if second_output.amount() < required_storage_deposit {
-            third_output_builder = third_output_builder.with_amount(required_storage_deposit)?;
+            let mut new_sdr_amount = required_storage_deposit - options.amount;
```

Comment:
Bit annoying to check in the diff but is `second_output.amount()` == `options.amount` ?

Reply:
It could be, but doesn't have to be

Reply:
`second_output.amount()` >= `options.amount`, so `second_output.amount() < required_storage_deposit` in 162 should take care of that

Reply:
Where is the `second_output.amount() >= options.amount`  guarantee?

Reply:
Oj because it's used to create the builder I assume

Reply:
It's very hard to follow if it's guaranteed, I'll trust you

Reply:
https://github.com/iotaledger/wallet.rs/blob/f5578dc0b71ab9e04c1cec1c3e7ec5197c7b79fa/wallet/src/account/operations/transaction/prepare_output.rs#L109-L111

Reply:
https://github.com/iotaledger/wallet.rs/blob/f5578dc0b71ab9e04c1cec1c3e7ec5197c7b79fa/wallet/src/account/operations/transaction/prepare_output.rs#L147